### PR TITLE
feat: port media input component to zui

### PR DESCRIPTION
**Pull request by ashmidgley**
[Port Media Component into zUI](https://www.notion.so/rational-nomads/Port-Media-Component-into-zUI-56a6a91909ed446d91cfb1b487b74bcb)

-----

## 1. Pull request checklist
- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
```
- Feature
```

## 3. What is the old behaviour?
No Media Input component available in zUI.

## 4. What is the new behaviour?
Media Input component added with stories and relevant tests.

### CODE REVIEW

File: src/components/MediaInput/MediaInput.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+});
+
+test('should render preview image when previewUrl valid and mediaType equals image', () => {
+  const { container } = renderComponent({
+    ...initialProps,
+    mediaType: 'image',
+    previewUrl: 'testing.png'
+  });
+
+  expect(container.querySelector('img')).toBeVisible();
```
Comment (colbr) [inline src/components/MediaInput/MediaInput.test.tsx:34]: Can we also assert that the `img` `src` is `previewUrl` (same for the video one below)?

File: src/components/MediaInput/MediaInput.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  mediaType: string | undefined;
+  previewUrl: string;
+  hasError: boolean;
+  onChange: (mediaType: string, previewImage: string, image: Buffer) => void;
+};
+
+const MediaInput: FC<MediaInputProps> = ({ mediaType, previewUrl, hasError, onChange }) => {
+  const inputFile = useRef<HTMLInputElement>(null);
+
+  const openUploadDialog = () => (inputFile.current ? inputFile.current.click() : null);
```
Comment (colbr) [inline src/components/MediaInput/MediaInput.tsx:14]: nit: curly braces here, and change to `inputFile.current?.click()`

**Review by colbr — CHANGES_REQUESTED** [review]
Looking good, nice job! Few requested changes but nothing crazy.

**Comment by ashmidgley** [comment]
Cheers for the review @colbr! Great calls all round. Have pushed changes to address the above.

**Review by colbr — COMMENTED** [review]
🚀

As discussed, change that `input.tsx` to `input.ts` and we're good!

**Review by colbr — APPROVED** [review]
🚀

**Pull request merged**